### ISSUE TITLE
Cancelled Events ignored on Bungeecord

### ISSUE THREAD

**Issue by WalrusSlayer** [games647/FastLogin#471]
[//]: # (Lines in this format are considered as comments and will not be displayed.)
[//]: # (Before reporting make sure you're running the **latest build** of the plugin and checked for existing issues!)

### What behaviour is observed:
[//]: # (What happened?)
The plugin is not capable of supporting connections against an attack with other plugins like 'AntiBot'.

Since FastLogin ignores cancelled events, it allows a bot raid to crash any FastLogin server with ease.
### What behaviour is expected:
[//]: # (What did you expect?)
It should support connections against attacks with other plugins.
### Steps/models to reproduce:
[//]: # (The actions that cause the issue. Please explain it in detail)
No other explanation or info is needed as it's an inherent flaw in the plugin itself.
### Plugin version or build number (don't write latest):
[//]: # (This can be found by running `/version plugin-name`.)
#964
### Server Log:
[//]: # (No images please - only the textual representation)
[Hastebin](https://hastebin.com/) / [Gist](https://gist.github.com/) link of the error, stacktrace or the complete log (if any)
https://paste.gg/p/Walrus2020/80701b08e3784e9cbbee2ec94acbcd56

Thanks!

**Comment by games647** [comment]
I already had this a lot of times. FastLogin ignores cancelled events. 

BungeeCord has a design that async events won't pause other events from firing. If your anti bot plugin performs an asynchronous operation (typically a database or web request), the event will still fire for other plugins. So plugins will still handle the request, while the anti bot plugin waits for response on a connection or similar. We only see the cancel result after the anti bot plugin is finished. So you have two flows working concurrently. Namely a race condition. In recent versions, FastLogin has a rate limiter, that completely ignores players if the rate hits. However it's very limited. Anti bit plugins or even better non-plugin solutions should handle this.

https://github.com/games647/FastLogin/blob/aa734853c6fe171d581523ef93710c4e952b8284/bungee/src/main/java/com/github/games647/fastlogin/bungee/listener/ConnectListener.java#L74

See GH-Ticket: https://github.com/games647/FastLogin/issues/304

TLDR from that ticket; Make the check synchronous or use a event that fires before it.

**Comment by games647** [comment]
Closing in favor of the other ticket. I already proposed to some anti plugin developers, that they could develop an open API, that FastLogin could listen to in order to wait for the anti bot check result. However progress stopped, in the end, some I noticed some plugins listen to FastLogin events and cancel them before FastLogin handles them.

**I'm still open to add that integration.**

**Comment by games647** [comment]
BTW: if a ticket is closed, you cloud still write in it. It just means it's marked as resolved. In your practical case you could also tune the config settings, to make the rate limiter stricter.
